src/index.ts

```diff
@@ -142,35 +142,36 @@ export function either<T>(parser: Parser<T>): Parser<{isError: boolean, value: T
   });
 };
 
-// coroutine :: (() -> Iterator (Parser e a s)) -> Parser e a s
-export function coroutine<T>(g: FnReturingParserIterator<T>): Parser<T> {
+// coroutine :: (() -> e) -> Parser e a s
+export function coroutine<T>(scanner: (tokenizer:<K>(parser:Parser<K>)=>K)=>T): Parser<T> {
```

Comment:
Maybe break this function type out into it's own declaration above.

Reply:
I added a seperate declaration for now but i would refactor all types in near future